pos_lot_selection/static/src/js/popup/ProductScreen.js

```diff
@@ -0,0 +1,46 @@
+/*
+    Copyright 2022 Camptocamp SA (https://www.camptocamp.com).
+    License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+*/
+odoo.define("pos_lot_selection.ProductScreen", function (require) {
+    "use strict";
+
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const Registries = require("point_of_sale.Registries");
+    const NumberBuffer = require("point_of_sale.NumberBuffer");
+
+    const PosLotSaleProductScreen = (ProductScreen) =>
+        class extends ProductScreen {
+            async _clickProduct(event) {
```

Comment:
Overriding `_clickProduct` won't provide this module's feature if the product barcode was scanned.. I feel like it would be better to override `_getAddProductOptions` or even `showPopup` instead.